### PR TITLE
AN-5116/sl-2-macros

### PR DESCRIPTION
**Pull request by drethereum**
1. Sets up fsc-evm repo as usage as a new DBT package
2. Adds Macros for EVM Streamline 2.0 upgrades and utility models
3. Updated README
4. Version tag v1.0.0

### CODE REVIEW

File: macros/streamline/silver/core/streamline_core_requests.sql [hunk 1]
```diff
@@ -0,0 +1,234 @@
+{% macro streamline_core_chainhead(
+        vault_secret_path
+    ) %}
+SELECT
+    live.udf_api(
```
Comment (austinFlipside) [inline macros/streamline/silver/core/streamline_core_requests.sql:5]: Do we need the `{{ target }}` for the udf?

File: macros/streamline/silver/core/streamline_core_requests.sql [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        'POST',
+        '{service}/{Authentication}',
```
Comment (austinFlipside) [inline macros/streamline/silver/core/streamline_core_requests.sql:7]: I think this should be a param as it won't be the same outside of QN. I also think we need to clean up the `s` in `service` vs. `A` in `Authentication`. The QN set up should be the default value though

File: macros/streamline/silver/core/streamline_core_requests.sql [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        '{{ vault_secret_path }}'
+    ) AS request
+        FROM
+            {% if model != 'confirmed_blocks' %}
+                ready_blocks
+            {% else %}
+                to_do
+            {% endif %}
+        ORDER BY
+            partition_key ASC
```
Comment (austinFlipside) [inline macros/streamline/silver/core/streamline_core_requests.sql:229]: I think filling in ascending order makes sense but it is something we should be aware we are deciding

File: macros/streamline/silver/core/streamline_core_retry.sql
Comment (austinFlipside) [inline macros/streamline/silver/core/streamline_core_retry.sql:1]: This looks good but it'll be interesting to see how this evolves with other chains.

File: macros/streamline/silver/core/streamline_core_requests.sql (hunk 3 above)
Comment (drethereum) [reply macros/streamline/silver/core/streamline_core_requests.sql:229]: we can make the order by a param as well

File: macros/streamline/silver/core/streamline_core_requests.sql (hunk 1 above)
Comment (drethereum) [reply macros/streamline/silver/core/streamline_core_requests.sql:5]: not necessary, we can add back if thats the preference

**Review by austinFlipside — APPROVED** [review]

**Pull request merged**